src/index.js

```diff
@@ -118,37 +129,27 @@ export class Circuit extends HeadlessCircuit {
             });
             graph.set('laid_out', true);
         }
-        paper.unfreeze({
-            progress(done, processed, total) {
-                if (done) {
-                    paper.fitToContent({ padding: 30, allowNewOrigin: 'any' });
-                    if (opts.onDone) opts.onDone();
-                }
-            }
+        this.listenTo(paper, 'render:done', function() {
+            paper.fitToContent({ padding: 30, allowNewOrigin: 'any' });
         });
+        paper.unfreeze();
         // subcircuit display
-        this.listenTo(paper, 'cell:pointerdblclick', function(view, evt) {
-            if (!(view.model instanceof cells.Subcircuit)) return;
+        this.listenTo(paper, 'open:subcircuit', function(model) {
             const div = $('<div>', { 
-                title: view.model.get('celltype') + ' ' + view.model.get('label') 
+                title: model.get('celltype') + ' ' + model.get('label')
             }).appendTo('html > body');
             const pdiv = $('<div>').appendTo(div);
-            const graph = view.model.get('graph');
-            var didResize = false;
-            const paper = this.makePaper(pdiv, graph, {
-                onDone() {
-                    if (didResize) return;
-                    didResize = true;
-                    const maxWidth = $(window).width() * 0.9;
-                    const maxHeight = $(window).height() * 0.9;
-                    div.dialog({ width: Math.min(maxWidth, pdiv.outerWidth() + 60), height: Math.min(maxHeight, pdiv.outerHeight() + 60) });
-                    div.on('dialogclose', function(evt) {
-                        paper.remove();
-                        div.remove();
-                    });
-                }
+            const graph = model.get('graph');
+            const paper = this.makePaper(pdiv, graph);
+            paper.once('render:done', function() {
+                circuit._windowCallback('Subcircuit', div, () => {
+                    div.remove();
```

Comment:
Is there a reason why there is no paper.remove() here? Without it, I worry that the paper will still listen to the circuit.

Reply:
Nice catch, I've added `paper.remove()` here.